charmhelpers/contrib/openstack/templates/memcached.conf

```diff
@@ -38,7 +38,7 @@ logfile /var/log/memcached.log
 # Specify which IP address to listen on. The default is to listen on all IP addresses
 # This parameter is one of the only security measures that memcached has, so make sure
 # it's listening on a firewalled interface.
--l {{ memcache_server }}
+-l {{ memcache_listen_address }}
```

Comment:
I'm fairly uncomfortable with this change. By doing this, the memcached instance is being opened to, potentially, the world.

A better solution, in my mind, would be to actually relate to a coordinator memcached[1] server to handle this, rather than make all units implicitly members of a globally usable memcached service.

[1]: Designate uses a [copordinator memcached](https://github.com/openstack/charm-designate/blob/master/src/tests/bundles/bionic-rocky.yaml#L68-L69) to handle sharing cache between units.

Reply:
Looking at how this is implemented I noticed that the `memcached` instances deployed by the memcached charm are just as open:

https://git.launchpad.net/charm-memcached/tree/templates/memcached.conf#n46

What am I missing?

(I am not opposed to your suggestion at all. I just would like to understand 😄 In fact I think to factor out the memcached logic into separate units is a great move regardless of the security implications).

Reply:
The memcached charm actually manages access control[]via ufw](https://git.launchpad.net/charm-memcached/tree/hooks/memcached_utils.py#n22) though, which the OpenStack Charm's internal memcached does not

Reply:
Oh nice! Thanks for pointing that out, I had missed that. I'll get to work then :smile:

Reply:
I have updated the change here and added a relation to the `memcached` charm. This change requires a change in `metadata.xml` on the charm side which I will submit separately.

Reply:
I will revert the changes to `memcached.conf` in another iteration of this PR.